### DISCUSSION TITLE
Support .Net Core 3.1, .Net 5 / 6

### DISCUSSION THREAD

**Discussion by talor1981** [dadhi/DryIoc#425] [Q&A]
Hi All,
Is this package already support the .Net Core 3.1 & .Net 5?
Is there any plan to upgrade this package to support in the future .Net 6?

**Comment by dadhi** [comment]
The first two are supported.
The .Net 6 is work in progress.
Some breaking changes in .Net 6 DI were introduced so I need to figure out how to support both older and newer version.

**Reply by maulik-modi** [reply]
@dadhi , Is DryIOC .NET 6 ready now?

**Reply by maulik-modi** [reply]
Could you please highlight .Net 6 DI breaking changes? .NET 3.1 is supported till end of December 2022 so, I understand your reasoning of supporting both.

**Reply by dadhi** [reply]
Actually I don't recall any API breaking changes. 
You still probably need to recompile to get the advantage  of the new features.

You may also use fuget.org to campare different versions of nuget packages (preview included).

**Comment by dadhi** [comment]
There are previews for both DryIoc 5 and https://www.nuget.org/packages/DryIoc.Microsoft.DependencyInjection/6.0.0-preview-02
I am planning to release the stable versions soon.

**Reply by HypeillingerChillinger** [reply]
Did you release .Net 6 already and is there a web api example implementation for .Net 6?

I found one for 3.1 https://github.com/dadhi/DryIoc/tree/master/samples/DryIoc.AspNetCore31.WebApi.Sample

**Reply by HypeillingerChillinger** [reply]
The link https://github.com/graftedbranch/dryiocwebapi.sample is dead by the way

found here: https://github.com/dadhi/DryIoc/blob/master/docs/DryIoc.Docs/Home.md Samples > External Links > DryIoc.WebApi sample

**Comment by dadhi** [comment]
@HypeillingerChillinger Thanls for finding the dead link. Will fix.

There are no full blown example app, but there are two barebone samples: MinimalWeb and DryIocDemo in the samples folder.

**Reply by HypeillingerChillinger** [reply]
Thanks for the hint. It works.

Very good project by the way. Another star for you... Thank you so much

**Reply by dadhi** [reply]
@HypeillingerChillinger I have fixed the docs in the v6-dev branch. Just for tracking it here.
